plugins/modules/rpm_sync.py

```diff
@@ -106,6 +126,18 @@ def main():
             mirror = module.params["sync_policy"] == "mirror_complete"
             parameters = {"mirror": mirror}
 
+        parameters = {"optimize": module.params["optimize"]}
```

Comment:
you are overwriting what we already collected in parameters here. This cannot possibly be correct.

Reply:
Good call out, I'll fix that. I also realized there's no test for the sync_policy key:pair. I'll proactively add that in.

Reply:
I guess you can fold that into the same update calls, no?

Reply:
Yep, exactly. I'll do that